src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
```

Comment:
lol, you could actually argue that it's more misleading to account for cases that are illegal. the fact that a bug slipped through, kinda spoke to the readability of the construct. 

I also found inspecting the orders kind of hard. When I noticed the bug, I looked at the test cases, assuming that we must have been missing the case the bug messed up. But it seems some of the test cases were also wrong. 

The way that the quantity is oriented towards the direction makes it a little harder to eyeball the `makerExchangeRate` for correctness. I'm not sure if it's just a habit, but I feel myself wanting to orient towards inputs/outputs and takerPays/takerGets.

A lot of this doesn't really feel right to me. To me the sell flag just defines when the order stops executing in the case when the maker gets a better exchange than what they proposed. If they wanted to get 500 XRP and give 1 USD, and they could make use of an offer giving 1000 XRP for 1 USD, do they give up their whole USD? or just 50 cents?

I'd like to see how the default grouping of bids/asks and inverseBids/inverseAsks is going to be used. To me, a more useful distinction for asks/bids is inputs(takerPays) and outputs(takerGets).  Admittedly, I'm coming from limited experience with (mostly just reading) order book display code.